pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -431,6 +437,80 @@ protected CompletableFuture<TransactionPendingAckInternalStats> internalGetPendi
+                .thenCompose(topic -> {
+                    TransactionBuffer.SnapshotType snapshotType = topic.getTransactionBuffer().getSnapshotType();
+                    if (snapshotType == null) {
+                        return FutureUtil.failedFuture(new RestException(NOT_FOUND,
+                                "Transaction buffer Snapshot for the topic does not exist"));
+                    } else if (snapshotType == TransactionBuffer.SnapshotType.Segment) {
+                        transactionBufferInternalStats.snapshotType = snapshotType.toString();
+                        this.topicName = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT_SEGMENTS);
+                        return getTxnSnapshotInternalStats(topicName, authoritative, metadata)
+                                .thenApply(snapshotInternalStats -> {
+                                    transactionBufferInternalStats.segmentInternalStats = snapshotInternalStats;
+                                    return transactionBufferInternalStats;
+                                }).thenCompose(ignore -> {
+                                    this.topicName = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                            SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT_INDEXES);
+                                    return getTxnSnapshotInternalStats(topicName, authoritative, metadata)
+                                            .thenApply(indexStats -> {
+                                                transactionBufferInternalStats.segmentIndexInternalStats = indexStats;
+                                                return transactionBufferInternalStats;
+                                            });
+                                });
+                    } else if (snapshotType == TransactionBuffer.SnapshotType.Single) {
+                        transactionBufferInternalStats.snapshotType = snapshotType.toString();
+                        this.topicName = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT);
+                        return getTxnSnapshotInternalStats(topicName, authoritative, metadata)
+                                .thenApply(snapshotInternalStats -> {
+                                   transactionBufferInternalStats.singleSnapshotInternalStats = snapshotInternalStats;
+                                   return transactionBufferInternalStats;
+                                });
+                    }
+                    return FutureUtil.failedFuture(new RestException(INTERNAL_SERVER_ERROR, "Unknown SnapshotType "
+                            + snapshotType));
+                });
+    }
+
+    private CompletableFuture<SnapshotInternalStats> getTxnSnapshotInternalStats(TopicName topicName,
```

Comment:
Could you add a test for this scenario?

Reply:
Oh, thanks for the reminder.
I add a `testGetTransactionBufferInternalStatsInMultiBroker` to cover the case of system topic existing in multiple brokers. And the case of a single broker had been covered by `testGetTransactionBufferInternalStats`.

Reply:
I think this method `getTxnSnapshotInternalStats` is used to get the info of ManagedLedger, can we rename this method and the class `SnapshotInternalStats `